### DISCUSSION TITLE
Reduce dependencies and size of packages like @scalar/hono-api-reference

### DISCUSSION THREAD

**Discussion by sebastianwessel** [scalar/scalar#2649] [Ideas]
Hi,
Thanks for providing such a nice tooling.
I like to use the `@scalar/hono-api-reference` as default and recommended UI for OpenAPI in my own project [PURISTA](https://purista.dev).
But, there are a bunch of dependencies loaded - including esbuild, storybook stuff.

From what I've seen, the heavy stuff comes from the `@scalar/components` dependency.

```json
"dependencies": {
    "@floating-ui/utils": "^0.2.2",
    "@floating-ui/vue": "^1.0.2",
    "@headlessui/vue": "^1.7.20",
    "@storybook/test": "^8.0.8",
    "@vueuse/core": "^10.10.0",
    "cva": "1.0.0-beta.1",
    "nanoid": "^5.0.7",
    "radix-vue": "^1.8.4",
    "tailwind-merge": "^2.3.0",
    "vue": "^3.4.29",
    "@scalar/code-highlight": "0.0.7"
  },
  ```
  
 Maybe you can fine tune the deps? As an example, `@storybook/test` requires storybook, which then requires esbuild...
 Imo, packages like `@scalar/hono-api-reference` should only contain the final built, bundled and optimized version of the UI, without any other deps.
 
 Installing `@scalar/hono-api-reference` results in round about 520 dependencies, with a size of around 230MB, which is simply a no-go, to use this package as it is.

**Comment by amritk** [comment]
Hey @sebastianwessel thanks for the awesome post! Improving performance is always a top priority. 

That being said I looked at the hono package and it looks like it uses our CDN build which has 0 dependencies. I'm not sure why the references package is even in there! I will remove it!

**Reply by sebastianwessel** [reply]
Does it only use the same build as the CDN - meaning the files are in the package and used from it.
Or, is it loading the files from the CDN - meaning, it requires having always an internet connection?

**Reply by amritk** [reply]
Internet connection, it loads the files from the cdn on load

**Reply by sebastianwessel** [reply]
Good to know. Because, it is ok for most use cases, but as soon as there is no internet connection, you simply can not use the UI.
Example: working in airplanes and trains. Also, there might be privacy concerns in some use cases.

**Reply by amritk** [reply]
Yep its a good use case and we have been discussing a path forward! Any thoughts on what you would prefer? If we include the actual package it would be too many dependencies as you stated. 

However if we just include the built standalone package would that work for you?

**Reply by amritk** [reply]
Alright got a PR up for the first issue

https://github.com/scalar/scalar/pull/2650

**Reply by sebastianwessel** [reply]
I think, the approach of including the standalone package is the best way.
The swagger-ui package does it too.
It's a bit tricky to get the (file) content of a package available in a web server, but it works.

**Reply by platform-kit** [reply]
Hey guys, did this change ever ship? (Shipping the hono package with the standalone scalar package, rather than relying on CDN)
